Runtime/CSharp/Bitcraft.StateMachine.UnitTests/HandWritten/States/TransitionState.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bitcraft.StateMachine.UnitTests.HandWritten.States
+{
+    public class TransitionState : StateBase
+    {
+        public TransitionState()
+            : base(HandWrittenStateTokens.TransitionStateToken)
+        {
+        }
+
+        protected override void OnInitialized()
+        {
+            base.OnInitialized();
+
+            RegisterActionHandler(HandWrittenActionTokens.TerminateAction, (d, cb) =>
+            {
+                var testContext = (Context as StateMachineTestContext);
+
+                Assert.AreEqual(5, testContext.TestStatus);
+
+                --testContext.TestStatus;
+                cb(HandWrittenStateTokens.EndStateToken);
+            });
+        }
+
+        protected override void OnEnter(StateEnterEventArgs e)
+        {
+            base.OnEnter(e);
+
+            if (e.TriggeringAction != HandWrittenActionTokens.InitDoneAction)
+            {
+                e.Redirect.TargetStateToken = HandWrittenStateTokens.TransitionTargetStateToken;

```

Comment:
Out of scope of this PR but I just noticed that `TargetStateToken` (in `Redirect`) is poorly therefore missleading. It should just be named `TargetState`.